addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -99,23 +111,63 @@ public static boolean contains(String tag) {
-                                        .setConfidence(Alert.CONFIDENCE_HIGH)
-                                        .setDescription(getString("desc"))
-                                        .setSolution(getString("soln"))
-                                        .setReference(getString("refs"))
-                                        .setEvidence(element.toString())
-                                        .setCweId(345) // CWE-345: Insufficient Verification of Data
-                                        // Authenticity
-                                        .setWascId(15) // Application Misconfiguration
-                                        .raise());
+                        element -> {
+                            String integrityHash = calculateIntegrityHash(msg, element, tree);
+                            String otherInfo =
+                                    integrityHash.isEmpty()
+                                            ? ""
+                                            : Constant.messages.getString(
+                                                    MESSAGE_PREFIX + "otherInfo", integrityHash);
+                            newAlert()
+                                    .setRisk(Alert.RISK_MEDIUM)
+                                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                                    .setDescription(getString("desc"))
+                                    .setSolution(getString("soln"))
+                                    .setReference(getString("refs"))
+                                    .setEvidence(element.toString())
+                                    .setCweId(345) // CWE-345: Insufficient Verification of Data
+                                    // Authenticity
+                                    .setWascId(15) // Application Misconfiguration
+                                    .setOtherInfo(otherInfo)
+                                    .raise();
+                        });
+    }
+
+    private String calculateIntegrityHash(HttpMessage msg, Element element, SiteMap tree) {
+        String src = element.getAttributeValue("src");
+        String integrityHash = null;
+        if (src != null) {
+            URI newURI = null;
+
+            try {
+                String newuri = msg.getRequestHeader().getURI().toString();
+                newURI = new URI(newuri);
+                newURI = newURI.resolve(src);
+                SiteNode node =
+                        tree.findNode(
+                                new org.apache.commons.httpclient.URI(newURI.toString(), true));
+                integrityHash =
+                        DigestUtils.sha384Hex(
+                                node.getHistoryReference()
+                                        .getHttpMessage()
+                                        .getResponseBody()
+                                        .toString());
+            } catch (URISyntaxException e) {
```

Comment:
This should also catch `IllegalArgumentException` in order to address circumstances where the hostname portion of the URL is malformed (contains a character that is invalid) [which we have a unit test for :wink:].

`            } catch (URISyntaxException | IllegalArgumentException e) {`

Reply:
Okay I'll make the changes